### PR TITLE
add checkstyle, add environment file and create web driver manager

### PR DESCRIPTION
**Pull request by Raymundogs**

### CODE REVIEW

File: src/main/java/core/MyWebDriverManager.java [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+                                                                 driverManagerType)
+            throws ClassNotFoundException, NoSuchMethodException,
+            InvocationTargetException, InstantiationException, IllegalAccessException {
+        if (myWebDriverManager == null) {
+            myWebDriverManager = new MyWebDriverManager(driverManagerType);
+            WebDriverManager.getInstance(driverManagerType).setup();
+            Class<?> driverClass =  Class.forName(driverManagerType.browserClass());
+            driver = (WebDriver) driverClass.getDeclaredConstructor().newInstance();
+            wait = new WebDriverWait(driver, Long.parseLong(getDotenv().get("EXPLICIT_WAIT_TIME")));
+        }
```
Comment (danielaSantaCruz) [inline src/main/java/core/MyWebDriverManager.java:60]: Maybe, you could also configure the window maximization here.

File: src/main/java/salesforce/ui/pages/HomePage.java [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+     * @throws ClassNotFoundException    when the class does not exist
+     */
+    public HomePage(final DriverManagerType driverManagerType) throws InvocationTargetException,
+            NoSuchMethodException, InstantiationException, IllegalAccessException,
+            ClassNotFoundException {
+        super(driverManagerType);
+    }
+
+    @Override
+    protected void waitForPageToLoad() {
```
Comment (danielaSantaCruz) [inline src/main/java/salesforce/ui/pages/HomePage.java]: You should not have empty methods

File: src/test/java/CaseTests.java [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+  }
+
+  public void LoginWithValidCredentials() {
+    driver.get(getDotenv().get("BASE_URL_CASE"));
+    driver.findElement(By.id("username")).sendKeys(getDotenv().get("SALESFORCE_USERNAME"));
+    driver.findElement(By.id("password")).sendKeys(getDotenv().get("SALESFORCE_PASSWORD"));
+    driver.findElement(By.id("Login")).click();
+  }
+
+  public void logout() {
```
Comment (danielaSantaCruz) [inline src/test/java/CaseTests.java:50]: There should be no logic in a test, so you should move all the methods (logout, clickOnElements, etc) to another file

File: src/test/java/base/BaseTest.java [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+public class BaseTest {
+    private WebDriver driver;
+    protected LoginPage loginPage;
+
+    @BeforeClass
+    public void setUp() throws ClassNotFoundException, InvocationTargetException,
+            NoSuchMethodException, InstantiationException, IllegalAccessException {
+        driver = getWebDriverManager(DriverManagerType.CHROME).getDriver();
+        driver.get("https://login.salesforce.com/");
+        driver.manage().window().maximize();
```
Comment (danielaSantaCruz) [inline src/test/java/base/BaseTest.java]: You could configure the window maximization in your class MyWebDriverManager

File: src/main/java/core/MyWebDriverManager.java [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+    }
+
+    /**
+     * Creates the object myWebDriverManager.
+     *
+     * @param newDriverManagerType an enum with the web driver's type
+     */
+    public MyWebDriverManager(final DriverManagerType newDriverManagerType) {
+        this.driverManagerType = newDriverManagerType;
+    }
```
Comment (danielaSantaCruz) [inline src/main/java/core/MyWebDriverManager.java:105]: You should add a method to quit the driver

File: src/main/java/core/MyWebDriverManager.java [hunk 6]
```diff
@@ -0,0 +1,85 @@
...
+     * Gets the web driver wait instance according to its type.
+     *
+     * @return the web driver wait
+     * @throws ClassNotFoundException when the class does not exist
+     * @throws InvocationTargetException when the target can not be invoked
+     * @throws NoSuchMethodException when the method is not found
+     * @throws InstantiationException when it is not possible to create an instance
+     * @throws IllegalAccessException when it can not be accessed
+     */
+    public WebDriverWait getWebDriverWait() throws ClassNotFoundException,
```
Comment (GustavoZHA) [inline src/main/java/core/MyWebDriverManager.java]: The throws may not be necessary

File: src/main/java/salesforce/ui/pages/BasePage.java [hunk 7]
```diff
@@ -0,0 +1,80 @@
...
+ * license agreement you entered into with Fundacion Jala
+ */
+
+package salesforce.ui.pages;
+
+import io.github.bonigarcia.wdm.config.DriverManagerType;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.support.PageFactory;
+import org.openqa.selenium.support.ui.WebDriverWait;
+
```
Comment (GustavoZHA) [inline src/main/java/salesforce/ui/pages/BasePage.java:16]: Delete blank line

File: src/test/java/CaseTests.java [hunk 8]
```diff
@@ -0,0 +1,129 @@
...
+import org.openqa.selenium.By;
+import org.openqa.selenium.JavascriptExecutor;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.chrome.ChromeDriver;
+import org.openqa.selenium.chrome.ChromeOptions;
+import org.openqa.selenium.support.ui.ExpectedConditions;
+import org.openqa.selenium.support.ui.WebDriverWait;
+import org.testng.annotations.AfterTest;
+import org.testng.annotations.BeforeTest;
+
```
Comment (GustavoZHA) [inline src/test/java/CaseTests.java:16]: Delete blank line

File: src/main/java/core/MyWebDriverManager.java (hunk 1 above)
Comment (Raymundogs) [reply src/main/java/core/MyWebDriverManager.java:60]: Done.

File: src/main/java/salesforce/ui/pages/HomePage.java (hunk 2 above)
Comment (Raymundogs) [reply src/main/java/salesforce/ui/pages/HomePage.java]: Done.

File: src/test/java/CaseTests.java (hunk 3 above)
Comment (Raymundogs) [reply src/test/java/CaseTests.java:50]: I will change that class to use the webDriverManager but I will work on that on my next PR.

File: src/test/java/base/BaseTest.java (hunk 4 above)
Comment (Raymundogs) [reply src/test/java/base/BaseTest.java]: Done.

File: src/main/java/core/MyWebDriverManager.java (hunk 5 above)
Comment (Raymundogs) [reply src/main/java/core/MyWebDriverManager.java:105]: Done.

File: src/main/java/core/MyWebDriverManager.java (hunk 6 above)
Comment (Raymundogs) [reply src/main/java/core/MyWebDriverManager.java]: The method I am implementing requires them. But I will work to change it to a try and catch on a future PR.

**Review by GustavoZHA — APPROVED** [review]

**Review by danielaSantaCruz — APPROVED** [review]

**Pull request merged**